### PR TITLE
Adding support to produce avro encoded messages

### LINKED ISSUE #181: Possibility to produce avro encoded message to schema regustry powered topic
Hello

Some tools allow to produce avro encoded messges to topic using schema from schema registry using JSON file as input message.

For example this https://github.com/tchiotludo/akhq

### PR DESCRIPTION
**Pull request by fabiojmendes**
As it turns out, most of the heavy lifting was already done by the schemaCache implementation. If you think this approach is fine I would like to add support for avro keys as well.

Fixes #181

### CODE REVIEW

File: cmd/kaf/produce.go [hunk 1]
```diff
@@ -46,6 +47,8 @@ func init() {
 	produceCmd.Flags().StringVar(&timestampFlag, "timestamp", "", "Select timestamp for record")
 	produceCmd.Flags().Int32VarP(&partitionFlag, "partition", "p", -1, "Partition to produce to")
 
+	produceCmd.Flags().IntVarP(&schemaID, "schema-id", "", -1, "Value schema id for avro messsage encoding")
```
Comment (birdayz) [inline cmd/kaf/produce.go:50]: what about changing this to --avro-schema-id ? i think this is more in line with other flags like --proto-*

**Review by birdayz — CHANGES_REQUESTED** [review]

**Comment by fabiojmendes** [comment]
No problem! I love the cli kaf provides. It is much more intuitive and easy to use compared to the default kafka-console-consumer/producer. But avro is a must have for me :)

I'll move on to the avro key support and see how that looks like. Do you think it would make sense to create an Interface for encoders? That way we can streamline the code on the producer side instead of having a if statement to check for each case.

**Comment by birdayz** [comment]
I agree with the interface. We would have two implementations already: protobuf and avro. If those two match the interface, we can have good confidence that the interface is well designed.

**Comment by eshepelyuk** [comment]
@fabiojmendes @birdayz can you plz provide detailled examples how to use it ?
I am not sure this code can close #181

**Comment by fabiojmendes** [comment]
@eshepelyuk you need a json message that matches your avro schema and the `avro-schema-id` flag to the corresponding schema in schema-registry . Note that you also need to have the schema-registry configured for this to work.

Here's an example
```
echo '{"id":1,"name":"User"}' | kaf produce --avro-schema-id 3 some.topic
```

My schema looks like this:
```json
{
  "type": "record",
  "name": "user",
  "fields": [
    { "name": "id", "type": "int" },
    { "name": "name", "type": "string" }
  ]
}
```

If you would like to use a file, you can just cat the file into this pipeline using one record per line:
```
cat file.json | kaf produce --avro-schema-id 3 some.topic
```

**Comment by eshepelyuk** [comment]
Thnx !
Are you planning a new release with this feature included soon ?

**Comment by birdayz** [comment]
i published a fresh release.

**Pull request merged**